src/components/boardComponents/useItemActions.jsx

```diff
@@ -132,6 +129,9 @@ export const useItemActions = () => {
     ({ snapshot }) => async () => {
```

Comment:
I think you can have the same effect but also deal with different sized item with something like this version:

```js
   ({ snapshot }) => async () => {
      const selectedItemList = await getSelectedItemList(snapshot);

      // Count number of elements
      const numberOfElements = selectedItemList.length;

      // Compute
      const minMax = { min: {}, max: {} };
      minMax.min.x = Math.min(...selectedItemList.map(({ x }) => x));
@@ -141,13 +141,13 @@ export const useItemActions = () => {
          ({ x, id }) => x + document.getElementById(id).clientWidth
        )
      );
      minMax.max.y = Math.max(...selectedItemList.map(({ y, id }) => y));

      // Changes starts here --------------
      let newX = minMax.min.x;

      batchUpdateItems(selectedItems, (item) => {
        const { clientWidth } = document.getElementById(item.id);
        const newItem = {
          ...item,
          x: newX,
          y: minMax.min.y,
        };
        newX += clientWidth
        return newItem;
      });
     // Changes ends here -----------
    },
    [getSelectedItemList, batchUpdateItems, selectedItems]
);
```